client/src/lib/src/utils/hash.ts

```diff
@@ -0,0 +1,11 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) 2023 Savoir-faire Linux. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as crypto from 'crypto'
+
+export const hashString = (text: string, length: number = 32): string => {
+  const hash = crypto.createHash('sha256')
```

Comment:
minor optimisation. Can we keep the crypto item accross calls? Like storing it somewhere? Would it provide a non negligible optimization, or is it completly useless?

Reply:
It cannot be done. First thing I tried.

I just tried again, in case I had missed something. It gives the following error:
```
rejected promise not handled within 1 second: Error [ERR_CRYPTO_HASH_FINALIZED]: Digest already called
extensionHostProcess.js:144
stack trace: Error: Digest already called
 ```

Reply:
yeah we could except if we have a good idea?
Removing on deactivate will break other open workspaces?

Reply:
The "embedded-documents" folder is in `context.storageUri` (not to be confused with `context.globalStorageUri`. Each workspace has its own: `The uri of a workspace specific directory in which the extension can store private state.`. So there should not be any interference when one is deleted. It might cause trouble if the user somehow manages to open the same workspace twice, but my attempts to do so failed.